File: app/src/main/java/net/ginteam/carmen/network/api/service/RatingService.java
package net.ginteam.carmen.network.api.service;

import net.ginteam.carmen.model.Rating;
import net.ginteam.carmen.model.ResponseModel;
import net.ginteam.carmen.model.filter.CreateRating;
import net.ginteam.carmen.network.api.ApiLinks;

import retrofit2.http.Body;
import retrofit2.http.POST;
import retrofit2.http.PUT;
import retrofit2.http.Path;
import rx.Observable;

/**
 * Created by vadik on 20.02.17.
 */

public interface RatingService {

    @POST(ApiLinks.CATALOG.SET_RATING)
    Observable<ResponseModel<Rating>> setRating(
            @Body CreateRating createRating
    );

    @PUT(ApiLinks.CATALOG.UPDATE_RATING)
    Observable<ResponseModel<Rating>> updateRating(
            @Path(ApiLinks.CATALOG.ID) int ratingId,
            @Body Rating rating
    );

}

File: app/src/main/kotlin/net/ginteam/carmen/kotlin/api/service/RatingService.kt
package net.ginteam.carmen.kotlin.api.service

import net.ginteam.carmen.kotlin.api.ApiSettings
import net.ginteam.carmen.kotlin.manager.ApiManager
import net.ginteam.carmen.kotlin.model.InitialRating
import net.ginteam.carmen.kotlin.model.RatingModel
import net.ginteam.carmen.kotlin.model.ResponseModel
import retrofit2.http.Body
import retrofit2.http.POST
import retrofit2.http.PUT
import retrofit2.http.Path
import rx.Observable

/**
 * Created by eugene_shcherbinock on 2/26/17.
 */

interface RatingService {

    @POST(ApiSettings.Catalog.CREATE_RATING)
    fun createRating(@Body rating: InitialRating): Observable<ResponseModel<RatingModel>>

    @PUT(ApiSettings.Catalog.UPDATE_RATING)
    fun updateRating(
            @Path(ApiSettings.Catalog.Params.ID) ratingId: Int,
            @Body rating: RatingModel
    ): Observable<ResponseModel<RatingModel>>

    companion object {
        fun create(): RatingService = ApiManager.retrofit.create(RatingService::class.java)
    }

}